scripts/pi-hole/php/groups.php

```diff
@@ -0,0 +1,831 @@
+<?php
+/* Pi-hole: A black hole for Internet advertisements
+*  (c) 2019 Pi-hole, LLC (https://pi-hole.net)
+*  Network-wide ad blocking via your own hardware.
+*
+*  This file is copyright under the latest version of the EUPL.
+*  Please see LICENSE file for your rights under this license. */
+
+require_once('auth.php');
+
+// Authentication checks
+if(isset($_POST['token']))
+{
+    check_cors();
+    check_csrf($_POST['token']);
+}
+else
+{
+    log_and_die("Not allowed!");
+}
+
+$reload = false;
+
+require_once("func.php");
+require_once("database.php");
+$GRAVITYDB = getGravityDBFilename();
+$db = SQLite3_connect($GRAVITYDB, SQLITE3_OPEN_READWRITE);
+
+function JSON_success($message = null)
+{
+    header('Content-type: application/json');
+    echo json_encode(array("success" => true, "message" => $message));
+}
+
+function JSON_error($message = null)
+{
+    header('Content-type: application/json');
+    $response = array("success" => false, "message" => $message);
+    if(isset($_POST['action']))
+    {
+        array_push($response, array("action" => $_POST['action']));
+    }
+    echo json_encode($response);
+}
+
+if($_POST['action'] == "get_groups")
+{
+    // List all available groups
+    try
+    {
+        $query = $db->query("SELECT * FROM \"group\";");
+        $data = array();
+        while(($res = $query->fetchArray(SQLITE3_ASSOC)) != false)
+        {
+            array_push($data,$res);
+        }
+        echo json_encode(array("data" => $data));
+    }
+    catch (\Exception $ex)
+    {
+        return JSON_error($ex->getMessage());
+    }
+}
+elseif($_POST['action'] == "add_group")
+{
+    // Add new group
+    try
+    {
+        $stmt = $db->prepare('INSERT INTO "group" (name,description) VALUES (:name,:desc)');
+        if(!$stmt)
```

Comment:
If your following the PSR-2 guidelines you should be formatting your if statements with a space before the conditional and the brace on the same line:
`
if (!$stmt) {
 throw new Exception();
}
`
https://www.php-fig.org/psr/psr-2/#11-example

Reply:
When we do it, we should do it right, on the page you linked, I read

> As of 2019-08-10 **PSR-2 has been marked as deprecated**.
> [PSR-12](https://www.php-fig.org/psr/psr-12/) is now recommended as an alternative.

Reply:
Yes, your correct I missed the deprecations.